device/isis/segmentrouting.yaml

```diff
@@ -0,0 +1,289 @@
+components:
+  schemas:
+    Isis.SegmentRouting:
+      description: |-
+        Segment Routing (SR) turns on any node to select any path (explicit or derived from IGPs SPT computations) 
+        for each of its traffic classes. The path does not depend on a hop-by-hop 
+        signaling technique (unlike LDP or RSVP). 
+        It only depends on a set of segments that are advertised by the IS-IS routing protocol. 
+        These segments act as topological sub-paths that can be combined together to form the required path.
+        Reference: https://datatracker.ietf.org/doc/html/rfc8667
+      type: object
+      properties:
+        router_capability:
+          description: |-
+            Optional IS-IS TLV named CAPABILITY, formed of multiple sub-TLVs, which allows a router to announce its
+            capabilities within an IS-IS level or the entire routing domain.        
+          $ref: '#/components/schemas/Isis.RouterCapability'
+          x-field-uid: 1
+        node_prefix:
+          description: >-
+            ISIS Router Node prefix.
+          $ref: './prefixattributes.yaml/#/components/schemas/IsisSR.RouterNodePrefix'
+          x-field-uid: 2
+  
+    IsisSR.RouterNodePrefix:
+      description: >-
+        SID/Index/Label: As defined in Section 2.1.1.1.
+        Reference: https://datatracker.ietf.org/doc/html/rfc8667#VANDLFLAGS.
+      type: object
+      properties:
+        choice:
+          description: |-
```

Comment:
Somehow I am not able to map this choice have`ipv4_te_router_id` and `interface_ip` within [Prefix Segment Identifier (Prefix-SID) Sub-TLV](https://datatracker.ietf.org/doc/html/rfc8667#name-prefix-segment-identifier-p)
Those are probably applicable only for `Isis.RouterCapability`.